pages/events/[...slug].vue

```diff
@@ -46,14 +34,26 @@ const speakerTalk = speakerId => {
   })
 }
 
+const getSpeakerName = speakerId => {
+  const speaker = speakers.find(speaker => speaker.speakerId === speakerId)
+  return speaker.name
+}
+const getSpeakerCompanyName = speakerId => {
+  const speaker = speakers.find(speaker => speaker.speakerId === speakerId)
+  return speaker.company
+}
```

Comment:
Why not extract the speaker object once and then use it across the file?

Reply:
the json restructure fixed it